docs/Targets.md

```diff
@@ -0,0 +1,341 @@
+# Targets in `oras-go` v2
+
+> [!IMPORTANT]
+> Prerequisite reading: [Modeling Artifact](./Modeling-Artifacts.md)
+
+In `oras-go` v2, artifacts are modeled as [Directed Acyclic Graphs (DAGs)](https://en.wikipedia.org/wiki/Directed_acyclic_graph) stored in [Content-Addressable Storages (CASs)](https://en.wikipedia.org/wiki/Content-addressable_storage). Each node in the graph represents a [descriptor](https://github.com/opencontainers/image-spec/blob/v1.1.1/descriptor.md) of the content, which must include the following three properties:
+
+- `mediaType`: The media type of the referenced content.
+- `digest`: The digest of the targeted content.
+- `size`: The size, in bytes, of the raw content.
+
+An example of a descriptor for an [OCI Image Manifest](https://github.com/opencontainers/image-spec/blob/v1.1.1/manifest.md) is shown below:
+
+```json
+{
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "size": 7682,
+  "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270"
+}
+```
+
+## Interfaces
+
+Based on the graph-modeling concepts and descriptors, the following major interfaces are defined in `oras-go` v2.
+
+### Storage
+
+The [`Storage`](https://pkg.go.dev/oras.land/oras-go/v2/content#Storage) interface represents a content-addressable storage (CAS) where content is accessed via descriptors. It provides the following functions:
+
+- `Fetch`: Retrieves the content identified by the descriptor from the CAS.
+- `Exists`: Check whether the described content is present in the CAS.
+- `Push`: Add content matching the expected descriptor to the CAS.
+
+For example, consider the following graph stored in a `Storage`, where node names are aliases for descriptors:
+
+```mermaid
+graph TD;
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+```
+
+The effects of the `Fetch` and `Exists` functions would be:
+
+```
+Fetch(m0) == content_m0
+
+Exists(b0) == true
+Exists(b3) == false
+```
+
+If a new blob `b3` is pushed to the storage, the graph would update as follows:
+
+```mermaid
+graph TD;
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+Blob3["Blob b3"]
+```
+
+#### GraphStorage
```

Comment:
Should this be ### instead of ####? Target and GraphTarget are both ###.

Reply:
Nice catch! I should change the `GraphTarget`  one then.